### PR TITLE
feat(app): insight FiltersSection UI — add/edit/delete filter predicates

### PR DESCRIPTION
**Pull request by youhaowei**
Add FiltersSection, FilterEditDialog and wire them into InsightConfigPanel so users can add, edit, reorder, and delete filter predicates on an insight.

## Changes

- **FiltersSection** — collapsible section mirroring MetricsSection; SortableList with inline Edit + Remove icons; stale-field references display as struck-through danger text (no crash); formatFilterValue renders `between` as `low … high`
- **FilterEditDialog** — add/edit dialog with field picker, operator picker, and value input; `between` shows two range inputs coalesced to `{ low, high }`; input type (text/number/date) auto-detected from field's ColumnType; key-reset pattern for clean state on re-open
- **InsightConfigPanel** — wires in FiltersSection + FilterEditDialog with `filtersWithIds` memoization, `handleFiltersReorder`, `handleRemoveFilter`, `handleSaveFilter`; persists via `updateInsight({ filters: [...] })` stripping client-only `_id` before save
- **index.ts** — exports FilterEditDialog, FiltersSection, FilterWithId

### FiltersSection with a filter added

**Light**

filters-light

**Dark**

filters-dark

### FilterEditDialog — edit mode (field + operator + value pre-populated)

filter-dialog-light

### FilterEditDialog — between operator (two range inputs)

filter-between-light

## Verification

- Navigated to insight page; FiltersSection appears below MetricsSection in config panel (left panel)
- Clicked "Add" in Filters section → FilterEditDialog opened with field picker, operator picker, value input
- Added filter `revenue_id eq 2024-01-01`; confirmed it appears in the Filters list with badge count 1
- Switched operator to `between`; confirmed two range inputs appear labeled "Low … High"
- Clicked edit icon on existing filter → dialog pre-populated with current values
- Deleted filter → removed immediately, no cascade/confirm dialog
- Reloaded page → filter persisted (survives round-trip through `updateInsight`)
- TypeCheck: 47/47 tasks pass

## Review round 2 — correctness fixes

Resolved all open review threads (3 Greptile P1 correctness bugs + 6 P2s).

**P1 — fixed**
- `in` + numeric field stored NaN: validation now rejects non-finite elements in an `in` list on a numeric field.
- Empty `in` saved silently (`","` → `IN ()`): empty parsed lists are now rejected.
- Index-based `_id` misrouted edits under a concurrent subscription: filters now carry a persisted stable `id` (generated on add, preserved through dexie + server round-trips); identity survives reorder, so saves can't misroute or duplicate.

**P2 — fixed**
- Dropped right join-keys and ambiguous duplicate column names are excluded from the filter-field picker (`computeFilterableFields`), so every offered field yields a working predicate.
- `in` membership filters are editable without destructive rewrite to a scalar.

**P2 — deferred (replied + resolved on-thread, tracked as follow-ups)**
- `hasConfiguration` counting filters-only insights — lives in `DataPreviewSection`; result-mode behavior change with its own surface.
- `contains` gating to text-compatible fields — V1 acceptable; refinement on the static operator list.
- Metrics as filter targets (HAVING) — needs its own picker UX; SQL layer already supports it.

**New tests**
- `filter-value.test.ts` — NaN/empty-`in`/between/scalar validation + value build.
- `filter-id.test.ts` — stable id survives a simulated concurrent reorder (+ regression guard for the old index scheme).
- `compute-combined-fields.test.ts` — dropped join-key and duplicate-column exclusion.

**Runtime verification** (app on :4300, real insight): `in`+numeric `"abc, 2024"` → Add disabled; all-comma `","` → Add disabled; `between` → two range inputs; delete → immediate (no confirm dialog); valid `"1, 2, 3"` → Add enabled.

<details open><summary><h3>Greptile Summary</h3></summary>

This PR adds full CRUD for insight filter predicates — `FiltersSection` (collapsible sortable list), `FilterEditDialog` (add/edit with operator/value inputs), and the wiring in `InsightConfigPanel`. It also ships `computeFilterableFields` to narrow the field picker to columns the SQL builder can actually resolve, and a stable-identity scheme (`filter.id` + `filter-id.ts`) that prevents concurrent subscription updates from misrouting in-flight edits.

- **Filter identity**: `InsightFilter.id` is generated at save-time (not dialog-mount) via `prepareFilterForSave`, so consecutive \"Add\" calls yield distinct filters rather than the second overwriting the first; `applyFilterSave` matches on `_id` (sourced from the persisted `id`) so a subscription-triggered reorder between open and save can't corrupt the target.
- **Validation**: `isFilterDraftValid` rejects empty `in` lists, non-finite numeric elements, and empty `between` bounds; `buildFilterValue` coerces numeric fields correctly; all paths are locked by unit tests.
- **Field narrowing**: `computeFilterableFields` applies a two-step filter — drop right join-keys by source table + column name (not bare name), then drop ambiguous duplicates — ensuring base columns that share a name with a dropped join key remain filterable.
</details>

<details><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge — the correctness bugs from round 1 are all resolved, identity logic is well-tested, and the only outstanding gap is a UX edge case (inverted between bounds) that doesn't corrupt stored data.

All three round-1 data-correctness bugs (NaN storage, empty-in, stale index identity) have been fixed and locked by unit tests. The stable-id scheme is sound: ids are generated per save, not per mount, so consecutive adds yield distinct filters; `applyFilterSave` matches by travelling id so concurrent reorders can't misroute edits. The only gap remaining is that `isFilterDraftValid` does not enforce `low ≤ high` for the `between` operator, which lets a user persist an always-false SQL predicate — a bad UX outcome but not data loss or corruption.

`filter-value.ts` — the `between` bounds validation is missing the low ≤ high ordering check.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx | New dialog for add/edit filter predicates; key-reset pattern correctly resets form state per session; `in` operator now included in OPERATOR_OPTIONS; save-time id generation via prepareFilterForSave prevents second-Add overwrite. |
| packages/app/src/app/insights/[insightId]/_components/config-panel/FiltersSection.tsx | New collapsible section mirroring MetricsSection; stale-field references shown with strikethrough; SortableList with inline edit/remove; reorder dispatches through stable _id-sourced items. |
| packages/app/src/app/insights/[insightId]/_components/config-panel/InsightConfigPanel.tsx | Wires FiltersSection + FilterEditDialog; filterableFields uses computeFilterableFields to exclude dropped join-keys and ambiguous duplicates from the picker; stripFilterIds correctly preserves persisted `id` while discarding client-only `_id`. |
| packages/app/src/app/insights/[insightId]/_components/config-panel/filter-id.ts | Pure helpers for stable client identity; deriveFilterId sources _id from persisted id (falling back to content+index for API-created filters); prepareFilterForSave generates id per-save (not per-mount) preventing consecutive-Add overwrite; applyFilterSave matches by _id not index. |
| packages/app/src/app/insights/[insightId]/_components/config-panel/filter-value.ts | Pure validation + value-building logic; in/NaN and empty-in bugs from round 1 are fixed; between bounds are validated for non-emptiness and finiteness but not for low <= high order, which can produce an always-false SQL predicate. |
| packages/app/src/lib/insights/compute-combined-fields.ts | New computeFilterableFields applies two-step exclusion: dropped right join-keys (matched by sourceTableId + columnName, not bare-name), then ambiguous duplicates; ordering ensures a base column sharing a name with a dropped join key remains filterable. |
| packages/core-dexie/src/db/schema.ts | Adds optional `id` field to the filter entity shape in the Dexie schema, enabling round-trip persistence of the stable filter identity. |
| packages/core-dexie/src/repositories/insights.ts | Maps the new `id` field in entityToInsight so persisted filter ids survive local DB round-trips. |
| packages/types/src/insights.ts | Adds optional `id` to InsightFilter with a clear doc comment explaining the stable-identity contract; fully backward-compatible. |

</details>

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant User
    participant FiltersSection
    participant InsightConfigPanel
    participant FilterEditDialog
    participant filter_id as filter-id.ts
    participant filter_value as filter-value.ts
    participant updateInsight

    User->>FiltersSection: click Add
    FiltersSection->>InsightConfigPanel: onAddClick()
    InsightConfigPanel->>InsightConfigPanel: setFilterToEdit("new")
    InsightConfigPanel->>FilterEditDialog: "filter="new""
    FilterEditDialog->>FilterEditDialog: "effectiveFilter = {_id: "__new__", ...}"
    FilterEditDialog->>FilterEditDialog: "mount FilterEditForm (key="__new__-new")"

    User->>FilterEditDialog: fill field/operator/value, click Save
    FilterEditDialog->>filter_value: isFilterDraftValid(draft)
    filter_value-->>FilterEditDialog: true
    FilterEditDialog->>filter_value: buildFilterValue(draft)
    filter_value-->>FilterEditDialog: coerced value
    FilterEditDialog->>filter_id: "prepareFilterForSave({_id:"__new__",...})"
    filter_id->>filter_id: genId() → "uuid-new"
    filter_id-->>FilterEditDialog: "{id:"uuid-new", _id:"uuid-new", ...}"
    FilterEditDialog->>InsightConfigPanel: onSave(FilterWithId)
    InsightConfigPanel->>filter_id: applyFilterSave(filtersWithIds, saved)
    filter_id-->>InsightConfigPanel: [...current, saved] (appended)
    InsightConfigPanel->>InsightConfigPanel: stripFilterIds (remove _id)
    InsightConfigPanel->>updateInsight: "{filters: [{id:"uuid-new", field, operator, value}]}"
    updateInsight-->>InsightConfigPanel: insight.filters updated
    InsightConfigPanel->>InsightConfigPanel: filtersWithIds recomputed via withFilterIds
    FilterEditDialog->>InsightConfigPanel: onOpenChange(false)
    InsightConfigPanel->>InsightConfigPanel: setFilterToEdit(null)
    FilterEditDialog->>FilterEditDialog: "effectiveFilter=null → FilterEditForm unmounts"
```
</details>

<sub>Reviews (7): Last reviewed commit: ["fix(app): don&#39;t duplicate an id-less fil..."](https://github.com/youhaowei/dashframe/commit/28d715f30dc0cfd405d6e5047f84da5e1dbdc61c) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=37420635)</sub>

### CODE REVIEW

File: packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+      return { low, high } satisfies InsightFilterBetweenValue;
+    }
+    if (isIn) {
+      // Parse comma-separated values; coerce to number if field is numeric
+      const items = scalarValue
+        .split(",")
+        .map((s) => s.trim())
+        .filter(Boolean);
+      return inputType === "number" ? items.map(Number) : items;
+    }
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx]: Fixed in a5e495c. \`isFilterDraftValid\` now rejects any non-finite element for a numeric \`in\` field: \`tokens.every((t) => isFinite(Number(t)))\`. The Save button gates on this, so `"abc, 2024"` on a numeric field can no longer persist `[NaN, 2024]`. Logic extracted to `filter-value.ts` and locked by a unit test (`rejects a non-finite element on a numeric field`). Runtime-verified: with field `new_mrr` + operator `in` + value `"abc, 2024"`, the Add button is disabled.

File: packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    if (!field) return false;
+    if (isBetween) {
+      if (betweenLow.trim() === "" || betweenHigh.trim() === "") return false;
+      if (inputType === "number")
+        return isFinite(Number(betweenLow)) && isFinite(Number(betweenHigh));
+      return true;
+    }
+    if (scalarValue.trim() === "") return false;
+    if (!isIn && inputType === "number") return isFinite(Number(scalarValue));
+    return true;
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx]: Fixed in a5e495c. After parsing the `in` list, an empty result is now rejected: `if (tokens.length === 0) return false;` in `isFilterDraftValid`. So `","` (or any all-whitespace-comma string) disables Save instead of persisting an always-false `IN ()`. Locked by unit tests (`rejects an all-comma input`, `rejects whitespace-only tokens`) and runtime-verified (all-comma input disables Add).

File: packages/app/src/app/insights/[insightId]/_components/config-panel/InsightConfigPanel.tsx [hunk 3]
```diff
@@ -129,6 +140,22 @@ export function InsightConfigPanel({
...
+   * dialog, so stale ids are never in flight. FilterEditDialog's "new" session
+   * generates a UUID for its own key-reset scope; that UUID is dropped here on
+   * the next derivation after save.
+   */
+  const filtersWithIds = useMemo((): FilterWithId[] => {
+    return (insight.filters ?? []).map((f, i) => ({
+      ...f,
+      _id: `${f.field}-${f.operator}-${i}`,
+    }));
+  }, [insight.filters]);
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/InsightConfigPanel.tsx]: Fixed in a5e495c. Filters now carry a persisted, stable `id` (added as optional on `InsightFilter`, generated on add by `FilterEditDialog`, preserved through the dexie entity round-trip and the server's opaque `setInsightFilter` store). The client `_id` is sourced from that persisted `id` (`deriveFilterId` in `filter-id.ts`), so a subscription firing mid-edit no longer shifts identities — `applyFilterSave` matches the edited filter by its travelling id, not its array index. Locked by a unit test that opens an edit on filter B, simulates a concurrent reorder, saves, and asserts B is updated in place (no duplicate); a companion regression-guard test documents that the old index-based scheme would misroute under the same reorder.

File: packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+              <SelectValue placeholder="Select a field" />
+            </SelectTrigger>
+            <SelectContent>
+              {combinedFields.length === 0 ? (
+                <div className="p-2 text-center text-sm text-neutral-fg-subtle">
+                  No fields available
+                </div>
+              ) : (
+                combinedFields.map((f) => (
+                  <SelectItem key={f.id} value={f.columnName ?? f.name}>
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx:203]: Fixed in a5e495c. New `computeFilterableFields` excludes right-side join keys (matched case-insensitively against `join.rightKey`, mirroring the SQL builder dropping them from the emitted subquery). The `FilterEditDialog` picker is fed this narrowed set, so a filter on a dropped join key can no longer be created. `FiltersSection` still receives the full field set, so an existing filter on such a field renders by name rather than as a stale reference. Locked by a unit test (`excludes the right-side join key`).

[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx:203]: Fixed in a5e495c. `computeFilterableFields` also excludes ambiguous duplicate column names — when base and joined tables share a column name, neither is offered, since the SQL resolver keys on the bare column name and cannot disambiguate them. This prevents saving a filter that resolves to the wrong table's column. Locked by a unit test (`excludes both sides of an ambiguous duplicate column name`). A fuller fix (resolving filters by field id so duplicates can be disambiguated rather than dropped) would require the SQL builder to change its resolution contract — noted as a follow-up, not in this PR.

File: packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+const OPERATOR_OPTIONS: { value: Operator; label: string }[] = [
+  { value: "eq", label: "equals (=)" },
+  { value: "ne", label: "not equals (≠)" },
+  { value: "gt", label: "greater than (>)" },
+  { value: "gte", label: "greater than or equal (≥)" },
+  { value: "lt", label: "less than (<)" },
+  { value: "lte", label: "less than or equal (≤)" },
+  { value: "contains", label: "contains" },
+  { value: "between", label: "between" },
+];
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx:60]: Fixed in a5e495c (also addressed Greptile's P1 on the same line). The `in` operator is now in `OPERATOR_OPTIONS` with a comma-separated value input; opening an existing membership filter renders its array as a comma-joined string (`initialScalar`), and saving re-parses to an array via `buildFilterValue`. So an API/agent-created `in` filter is no longer destructively rewritten to a scalar when edited. Numeric `in` fields coerce each element to a number and reject non-finite values. Locked by unit tests for both build (`builds a numeric array`, `builds a trimmed string array`) and validation.

File: packages/app/src/app/insights/[insightId]/_components/config-panel/InsightConfigPanel.tsx [hunk 6]
```diff
@@ -325,6 +386,16 @@ export function InsightConfigPanel({
...
         />
+
+        {/* Filters Section */}
+        <FiltersSection
+          filters={filtersWithIds}
+          combinedFields={combinedFields}
+          onReorder={handleFiltersReorder}
+          onRemove={handleRemoveFilter}
+          onEditClick={setFilterToEdit}
+          onAddClick={() => setFilterToEdit("new")}
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/InsightConfigPanel.tsx:407]: Acknowledged, deferring out of this PR's scope. `hasConfiguration` lives in `DataPreviewSection` (not touched here) and treating a filters-only insight as a queryable result mode is a preview/result-mode behavior change with its own surface (the result toggle, `useInsightPagination`, the Join-vs-result preview switch). This PR delivers the filter editor; wiring filters into the result-mode gate is a separate, reviewable change. Filing as a follow-up rather than expanding this PR. Resolving the thread to reflect the deliberate scope decision.

File: packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+// ============================================================================
+
+const OPERATOR_OPTIONS: { value: Operator; label: string }[] = [
+  { value: "eq", label: "equals (=)" },
+  { value: "ne", label: "not equals (≠)" },
+  { value: "gt", label: "greater than (>)" },
+  { value: "gte", label: "greater than or equal (≥)" },
+  { value: "lt", label: "less than (<)" },
+  { value: "lte", label: "less than or equal (≤)" },
+  { value: "contains", label: "contains" },
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx:57]: Acknowledged, deferring. Gating `contains` to text-compatible fields (or making the SQL `LIKE` cast-safe for non-string columns) is a validation refinement on top of the static operator list, which is acceptable for V1. The user-facing impact is a preview/query error on a misapplied `contains`, not data corruption. Filing as a follow-up to either hide `contains` for non-text fields or cast the predicate safely in the engine. Resolving as a deliberate V1 scope boundary.

File: packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+            <SelectContent>
+              {combinedFields.length === 0 ? (
+                <div className="p-2 text-center text-sm text-neutral-fg-subtle">
+                  No fields available
+                </div>
+              ) : (
+                combinedFields.map((f) => (
+                  <SelectItem key={f.id} value={f.columnName ?? f.name}>
+                    {f.displayName}
+                  </SelectItem>
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx]: Acknowledged, deferring. Offering metric output aliases as filter targets (routed to HAVING) needs its own UX — a distinct target group in the picker, aggregate-aware value handling, and the HAVING-vs-WHERE distinction surfaced to the user. The SQL layer already supports it; the editor V1 covers dimension filters. An existing metric-alias filter from the API path currently renders as a stale reference in `FiltersSection` (by name, not crashing). Filing as a follow-up for metric-filter UI. Resolving as deliberate V1 scope.

**Comment by youhaowei** [comment]
## Screenshots — FiltersSection UI (in real app context)

The insight FiltersSection, sibling to Fields/Metrics in the config panel.

**The section (light + dark):**
filters light
filters dark

**Edit dialog (field / operator / value — value input typed by field; date field → date picker):**
filter dialog

**Between operator (two range inputs):**
between

Note: editing uses a **modal dialog** (Edit filter), mirroring the config panel's MetricEditorModal pattern rather than pure-inline edit — both patterns exist in the panel; modal was chosen for the field/operator/value form.

V1 scope: basic typed value inputs (text/number/date by field type). Deferred to follow-up: data-driven categorical dropdowns, metrics-as-filter-targets (HAVING UI), contains-on-non-text gating.

File: packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+   * round-trips and InsightConfigPanel can match the saved filter by a stable
+   * identity rather than a fragile array index.
+   */
+  const newFilterId = useMemo(
+    () => crypto.randomUUID(),
+    // Intentionally empty: we want one UUID per mount of FilterEditDialog,
+    // not per render. A new session (filter null→"new") unmounts+remounts this
+    // component because the parent sets filterToEdit=null on close first.
+    [],
+  );
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/app/insights/[insightId]/_components/config-panel/FilterEditDialog.tsx]: Fixed in 656e1fa0 (head). You're right — `FilterEditDialog` is permanently mounted in `InsightConfigPanel` (only its `filter` prop changes), so the mount-scoped `useMemo(() => crypto.randomUUID(), [])` handed every Add the same id, and the second filter overwrote the first via `applyFilterSave`. The id is now generated **per save** in `prepareFilterForSave` (filter-id.ts), keyed on an explicit `NEW_FILTER_ID` sentinel: a genuine new draft gets a fresh id (and matching `_id`); an existing row keeps its `_id` and only backfills a persisted id. The stale "parent unmounts/remounts" comment is gone. Verified at runtime: two consecutive Adds persist as two distinct filters (count 1→2→3, all rows distinct, survives reload); editing updates in place without duplicating. Unit test: `two consecutive Adds yield distinct ids — second does NOT overwrite first`.

File: packages/app/src/lib/insights/compute-combined-fields.ts [hunk 10]
```diff
@@ -127,3 +127,51 @@ export function computeCombinedFields(
...
+  for (const f of combinedFields) {
+    const key = (f.columnName ?? f.name).toLowerCase();
+    colNameCounts.set(key, (colNameCounts.get(key) ?? 0) + 1);
+  }
+
+  return combinedFields.filter((f) => {
+    const colNameLower = (f.columnName ?? f.name).toLowerCase();
+    if (droppedRightKeysLower.has(colNameLower)) return false;
+    if ((colNameCounts.get(colNameLower) ?? 0) > 1) return false;
+    return true;
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/lib/insights/compute-combined-fields.ts]: Fixed in 656e1fa0. The exclusion is now scoped to the join's **right table** — a field is dropped only when `sourceTableId === join.rightTableId && columnName === join.rightKey`, not when its bare name matches any `rightKey`. Duplicate-name ambiguity is computed **after** that drop, so a base column whose name collides with a dropped joined key stays filterable (it's the only one of that name left in the emitted query). Regression test added: base `id` + joined `id` + `rightKey="id"` → only the joined-side `id` excluded, base `id` remains.

File: packages/app/src/lib/insights/compute-combined-fields.test.ts [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+      {
+        type: "inner",
+        rightTableId: "t2",
+        leftKey: "fk",
+        rightKey: "id", // lower-case, field is "ID"
+      },
+    ];
+    const result = computeFilterableFields(fields, joins);
+    expect(result.some((f) => f.columnName === "ID")).toBe(false);
+  });
```
[the remark this answers is not in the document]
Comment (youhaowei) [reply packages/app/src/lib/insights/compute-combined-fields.test.ts:88]: Done in 656e1fa0. Added `keeps a base column whose name matches a dropped joined join-key` in compute-combined-fields.test.ts: base `id` + joined `id` with `rightKey="id"` asserts the joined-side `id` is excluded and the base `id` remains filterable (and is the sole `id`, hence unambiguous).
